### PR TITLE
Create ADR template with first decision: Secure developer package supply

### PR DESCRIPTION
**Pull request by shanepeckham**
## Description
Closes #110

## Type of Change

- [X] Non-module change (e.g. CI/CD, documentation, etc.)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates.
  - [ ] Breaking changes.
  - [X] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings
- [X] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

**Comment by Nepomuceno** [comment]
Should we keep all the ADR in one file or create a folder for the ARD and put each decision in one file? just worried this file could get really big.

File: adr.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+<!-- This is an optional element. Feel free to remove. -->
+### Positive Consequences
+
+* Allow for flexibility while providing options.
+
+<!-- This is an optional element. Feel free to remove. -->
+### Negative Consequences
+
+* If not implemented, a customer's development packages can pose a security risk if the provenance and package is not vetted.
```
Comment (noodlefrenzy) [inline adr.md:51]: If _what_ not implemented? I would go into a bit more detail here that a more robust solution would need to be implemented for a customer to take it to production.

**Review by noodlefrenzy — CHANGES_REQUESTED** [review]
Little comment on the content since much of it stems from the discussion in Teams, mostly commenting on how to ensure you get the right structure for mADRs in place.

File: adr.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+Customers need a secure supply package solution for developers. The AIRI implementation aims to be secure by default but offer flexibility to conform with a customer's chosen approach to secure supply for developer packages.
+
+<!-- This is an optional element. Feel free to remove. -->
+## Decision Drivers
+
+* The solution must offer a secure but flexible approach to package supply for developers.
+
+## Considered Options
+
+* Containerize solutions all solutions built on AIRI. Container image scanning will be required and customers may already have a solution in place.Conforming to the first principles of the AIRI project, we do not want to add anything that is not mandatory and reduces flexibility for brownfield deployments.
```
[the remark this answers is not in the document]
Comment (shanepeckham) [reply adr.md:33]: Implemented

**Comment by shanepeckham** [comment]
> Should we keep all the ADR in one file or create a folder for the ARD and put each decision in one file? just worried this file could get really big.

For now we will, but we can look to create a new one per milestone, period etc. Let's solve that later

**Comment by noodlefrenzy** [comment]
@Nepomuceno we should be breaking them out into their own directory - the Markdown ADR repo (where I advised Shane to get the template) has a good demonstration of the structure: https://github.com/adr/madr/tree/3.0.0-beta.2/docs/decisions

**Comment by noodlefrenzy** [comment]
Basically, create a ~/docs/decisions/ or ~/docs/adrs/ directory, and store all ADRs under there. Each decision is its own file, versioned. Not only that, but even if you reconsider a previous decision, you _never_ alter the original file once it's been decided other than to mark it as deprecated and point it to a new file with the revised decision.

e.g. :
- 0005-tabs-vs-spaces-use-tabs.md
- ...
- 0008-tabs-vs-spaces-use-spaces-instead.md
(you'd create a new file with the revised decision, and change the status on the previous one)

File: adr.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
+---
+# Configuration for the Jekyll template "Just the Docs"
+parent: Decisions
+nav_order: 100
+title: ADR Template
+
+# These are optional elements. Feel free to remove any of them.
```
Comment (martinpeck) [inline adr.md:7]: should remove these

File: adr.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+#### Positive Consequences
+
+A central artefact management system would allow for central build and deployment of surrounding AIRI solutions. This would benefit the CI/CD pipeline for internal development.
+
+#### Negative Consequences
+
+Conforming to the first principles of the AIRI project, we do not want to add anything to the customer facing solution that is not mandatory and reduces flexibility for brownfield deployments.
+
+### Allow outbound access from AML/AI Studio
```
Comment (martinpeck) [inline adr.md:88]: Suggest moving this to the top, so that the chosen option is first

File: adr.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+## Decision Drivers
+
+* The solution must offer a secure but flexible approach to package supply for developers.
+
+## Considered Options
+
+* Containerize all solutions
+* Pre-download the packages for solutions
+* Use a artefact management solution
+* Allow outbound access from AML/AI Studio
```
Comment (martinpeck) [inline adr.md:38]: suggest moving this to the top so the chosen option is first

**Comment by martinpeck** [comment]
Agree with comments above regarding the creation of a folder structure.
These files shouldn't sit in the root.

File: adr.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
+---
+# Configuration for the Jekyll template "Just the Docs"
```
[the remark this answers is not in the document]
Comment (shanepeckham) [reply adr.md:2]: Addressed

File: adr.md (hunk 3 above)
Comment (shanepeckham) [reply adr.md:7]: Removed

File: adr.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+## Considered Options
+
+* Containerize all solutions
+* Pre-download the packages for solutions
+* Use a artefact management solution
+* Allow outbound access from AML/AI Studio
+
+## Decision Outcome
+
+Allow outbound access from AML/AI Studio but include in the documentation that this will need to be disabled off for all environments.
```
[the remark this answers is not in the document]
Comment (shanepeckham) [reply adr.md:42]: Corrected

File: adr.md (hunk 4 above)
Comment (shanepeckham) [reply adr.md:88]: Moved

File: adr.md (hunk 5 above)
Comment (shanepeckham) [reply adr.md:38]: Moved

File: adr.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (shanepeckham) [reply adr.md:2]: Removed

**Pull request merged**